compiler/luci/service/src/Nodes/CircleReshape.test.cpp

```diff
@@ -39,25 +39,25 @@ TEST(CloneNodeTest, clone_Reshape)
   ASSERT_EQ(node_reshape->newShape()->dim(1), cloned_reshape->newShape()->dim(1));
 }
 
-TEST(ShapeRuleTest, reshape_by_input_const_static)
+TEST(ShapeRuleTest, reshape_by_circle_const)
 {
   auto g = loco::make_graph();
   auto node_reshape = g->nodes()->create<luci::CircleReshape>();
   auto tensor_input = g->nodes()->create<luci::CircleInput>();
-  auto shape_by_input = g->nodes()->create<luci::CircleConst>();
+  auto shape_input = g->nodes()->create<luci::CircleConst>();
 
   tensor_input->dtype(loco::DataType::S32);
   tensor_input->shape({2, 3, 4});
   tensor_input->shape_status(luci::ShapeStatus::VALID);
 
-  shape_by_input->dtype(loco::DataType::S32);
-  shape_by_input->size<loco::DataType::S32>(2);
-  shape_by_input->at<loco::DataType::S32>(0) = 6;
-  shape_by_input->at<loco::DataType::S32>(1) = 4;
-  shape_by_input->shape_status(luci::ShapeStatus::VALID);
+  shape_input->dtype(loco::DataType::S32);
+  shape_input->size<loco::DataType::S32>(2);
+  shape_input->at<loco::DataType::S32>(0) = -1;
+  shape_input->at<loco::DataType::S32>(1) = 4;
+  shape_input->shape_status(luci::ShapeStatus::VALID);
 
   node_reshape->tensor(tensor_input);
-  node_reshape->shape(shape_by_input);
+  node_reshape->shape(shape_input);
```

Comment:
Why is this test modified? 

If you'd like to add `shape = circle_const(-1, 4)` case, Why didn't you add just new test?

Reply:
> Why is this test modified?
> 
> If you'd like to add `shape = circle_const(-1, 4)` case, Why didn't you add just new test?

I tried to make all the cases to test "dynamic" shape inference so that it can be related to this PR's changes.